include/seqan3/core/type_traits/all.hpp

```diff
@@ -6,21 +6,18 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides various type traits.
- * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
- */
-
-/*!\defgroup type_traits Type Traits
- * \brief Provides various type traits and their shortcuts.
- * \ingroup core
+ * \brief [DEPRECATED] Provides various type traits.
+ * \deprecated This header is deprecated and will be removed in SeqAn-3.1.
+ *             Please \#include <seqan3/utility/type_traits/all.hpp> instead.
  */
 
 #pragma once
 
-#include <seqan3/core/type_traits/pre.hpp>
-#include <seqan3/core/type_traits/basic.hpp>
-#include <seqan3/core/type_traits/deferred_crtp_base.hpp>
-#include <seqan3/core/type_traits/iterator.hpp>
-#include <seqan3/core/type_traits/range.hpp>
-#include <seqan3/core/type_traits/template_inspection.hpp>
-#include <seqan3/core/type_traits/transformation_trait_or.hpp>
+#include <seqan3/core/detail/deferred_crtp_base.hpp>
+#include <seqan3/core/detail/iterator_traits.hpp>
+#include <seqan3/core/detail/template_inspection.hpp>
+#include <seqan3/core/range/type_traits.hpp>
+#include <seqan3/utility/type_traits/all.hpp>
+
+SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1. "
+                         "Please #include <seqan3/utility/type_traits/all.hpp> instead.")
```

Comment:
Please leave the one liner even though it exceeds the line length. If you break the line here, unfortunately the second line is not properly displayed to the user if triggered.

Same for all other files

Reply:
Oops, I didn't know... I'll change it!